R/plot.R

```diff
@@ -0,0 +1,449 @@
+#' @param data Data frame with the individual values and density ratio estimates
+#' @param var Name of the variable to be plotted
+#' @param y_lab Name of the y-axis label ("Density Ratio" or "Log Density Ratio")
+#' @param sample.facet Logical indicating whether to facet the plot by sample. Default is TRUE.
+#'
+#' @return
+#'
+individual_uni_plot <- function(data, var, y_lab, sample.facet = TRUE){
+
+  y_max <- max(2,data$dr)
+  y_min <- min(-2, data$dr)
+
+  plot <-
+    ggplot(data, aes(x = .data[[var]], y = dr)) +
+    geom_point(aes(col = sample),
+               alpha = .6) +
+    theme_bw() +
+    labs(title = "Scatter plot of individual values and density ratio",
+         color = "Sample",
+         y = y_lab) +
+    geom_hline(yintercept = 0, linetype = "dashed")+
+    scale_colour_manual(values = c("firebrick", "steelblue"),
+                      breaks = c("numerator", "denominator"),
+                      labels = c("Numerator", "Denominator")) +
+    scale_y_continuous(limits = c(y_min, y_max))
+
+  if(sample.facet){
+    plot <- plot + facet_wrap(~sample)
+  }
+
+  return(plot)
+}
+
+#' Scatter plot of density ratios and individual variables
+#'
+#' A scatter plot showing the relationship between estimated density ratios and individual variables.
+#'
+#' @inheritParams dr.histogram
+#' @param vars Character vector of variable names to be plotted.
+#' @param output Character indicating whether output should be a list of individual plots ("individual"), or one facetted plot with all variables ("assembled"). Defaults to "individual".
+#' @param sample.facet Logical indicating whether to facet the plot by sample, i.e, showing plots separate for each sample, and side to side. Defaults to FALSE.
+#' @param nrow Integer indicating the number of rows in the assembled plot. If NULL, the number of rows is automatically calculated.
+#' @return Scatter plot of density ratios and individual variables.
+#' @export
+#'
+#' @examples
+plot_univariate <- function(object, vars, samples = "both", logscale = TRUE,
+                            output = "individual", sample.facet = FALSE,
+                            nrow = NULL) {
+
+  # Check object type
+  check.object.type(object)
+
+  # Create data object
+  data <- rbind(object$df_numerator, object$df_denominator)
+
+  # Check names in data and variable names
+  check.overriden.names(data)
+  check.var.names(vars, data)
+
+  # Estimate density ratio
+  data$dr <- predict(object, newdata = data)
+
+  # Creta sample identifier
+  data$sample <- rep(c("numerator", "denominator"),
+                     c(nrow(object$df_numerator), nrow(object$df_denominator)))
+
+  # Create a object selection variable (both, numerator, denominator)
+  obsselect <- match.arg(samples, c("both", "numerator", "denominator"))
+
+  if (obsselect != "both") {
+    data <- filter(data, sample == obsselect)
+  }
+
+
+  if (logscale) {
+
+    if(any(data$dr <= 0)){
```

Comment:
I think the code is neater indeed, but in this case the warning is printed even when there's no problem at all i.e., no converted values. I think the warning should only be printed if there's negative estimates, so I'd keep if(any(ext$dr <0) before the new code.

Reply:
Yes, you're right, but you can just wrap the warning in the following. The point was more that there is no need to have the `ext$dr <= tol` repeatedly in there (and to trim everything smaller than tol to tol, instead of everything smaller than 0). 
```
if(any(negdr)) {
  warning(...)
}
```

Reply:
Ah ok, I see. Did that and applied it everywhere! Also added tol as an argument to all functions